src/core/cgroup.c

```diff
@@ -1689,7 +1689,7 @@ static bool unit_get_needs_bpf_foreign_program(Unit *u) {
         if (!c)
                 return false;
 
-        return !LIST_IS_EMPTY(c->bpf_foreign_programs);
+        return c->bpf_foreign_programs;
```

Comment:
i find this hard to read btw. i.e. if i read this changed line i'd assume this would return an `BPFProgram*` object, but it actually returns a boolean

Reply:
Yeah, I was thinking about this too. I changed it to use `!!`.